packages/network-controller/src/NetworkController.ts

```diff
@@ -334,6 +345,7 @@ export type NetworkId = `${number}`;
  * @property networkConfigurations - the full list of configured networks either preloaded or added by the user.
  */
 export type NetworkState = {
+  selectedNetworkClientId: NetworkClientId;
```

Comment:
Considering that the registry is divided into Infura network clients and custom network clients, it might be useful for TypeScript reasons to also store a `selectedNetworkClientType` in addition to a `selectedNetworkClientId`. Otherwise, when you implement the method to get the selected network client, you'll have to do something like:

``` typescript
function isBuiltInNetworkClientId(id: string): type is BuiltInNetworkClientId {
  return Object.keys(InfuraNetworkType).includes(id);
}

class NetworkController {
  // ...

  getSelectedNetworkClient() {
    const networkClientId = this.state.selectedNetworkClientId;
    const networkClientType =
      isBuiltInNetworkClientId(networkClientId) ?
        NetworkClientType.Infura :
        NetworkClientType.Custom;
    return this.#autoManagedNetworkClientRegistry[networkClientType][networkClientId]
  }
}
```

whereas if you have a type you can just say:

``` typescript
class NetworkController {
  // ...

  getSelectedNetworkClient() {
    const networkClientId = this.state.selectedNetworkClientId;
    const networkClientType = this.state.selectedNetworkClientType;
    return this.#autoManagedNetworkClientRegistry[networkClientType][networkClientId];
  }
}
```

Reply:
makes sense, though IMO it would be best to have the `type` stored in the network configuration - and infura networks should be included the the set of network configurations.

Reply:
Kinda seems better to me as-is, as it has minimal state. Adding a type (either to top-level state or the network config) would make future state migrations more complicated (as we eventually remove the globally selected network and the concept of a built-in type).